src/common/com/intellij/plugins/haxe/util/HaxeResolveUtil.java

```diff
@@ -611,6 +611,10 @@ private static HaxeClass tryResolveClassByQNameWhenGetQNameFail(@NotNull PsiElem
       String nameWithPackage = getQName(fileChildren, name, true);
       result = findClassByQName(nameWithPackage, type.getContext());
     }
+    else {
+      name = tryResolveFullyQualifiedHaxeReferenceExpression(type);
```

Comment:
I don't fully get logic behind this, 

My implementation `tryResolveFullyQualifiedHaxeReferenceExpression` for #39 seems to be too common, but check for '.' I think should eliminate checks for some expressions, like `ArrayUtils.delete_if`